packages/dev/core/src/FrameGraph/frameGraphRenderContext.ts

```diff
@@ -0,0 +1,274 @@
+// eslint-disable-next-line import/no-internal-modules
+import type { Nullable, AbstractEngine, DrawWrapper, IColor4Like, Layer, FrameGraphTextureHandle, Effect, FrameGraphTextureManager, RenderTargetTexture } from "core/index";
+import { Constants } from "../Engines/constants";
+import { EffectRenderer } from "../Materials/effectRenderer";
+import { CopyTextureToTexture } from "../Misc/copyTextureToTexture";
+import { FrameGraphContext } from "./frameGraphContext";
+import { backbufferColorTextureHandle, backbufferDepthStencilTextureHandle } from "./frameGraphTypes";
+
+/**
+ * Frame graph context used render passes.
+ * @experimental
+ */
+export class FrameGraphRenderContext extends FrameGraphContext {
+    private readonly _effectRenderer: EffectRenderer;
+    private _currentRenderTargetHandle: FrameGraphTextureHandle;
+    private _debugMessageWhenTargetBound: string | undefined;
+    private _debugMessageHasBeenPushed = false;
+    private _renderTargetIsBound = true;
+    private readonly _copyTexture: CopyTextureToTexture;
+
+    /** @internal */
+    constructor(
+        private readonly _engine: AbstractEngine,
+        private readonly _textureManager: FrameGraphTextureManager
+    ) {
+        super();
+        this._effectRenderer = new EffectRenderer(this._engine);
+        this._copyTexture = new CopyTextureToTexture(this._engine);
+        this._currentRenderTargetHandle = backbufferColorTextureHandle;
+    }
+
+    /**
+     * Checks whether a texture handle points to the backbuffer's color or depth texture
+     * @param handle The handle to check
+     * @returns True if the handle points to the backbuffer's color or depth texture, otherwise false
+     */
+    public isBackbuffer(handle: FrameGraphTextureHandle): boolean {

```

Comment:
why do we need all these functions? Just to factorize? can we just not call the texture manager functions directly?